src/plugins/objectMigration/plugin.js

```diff
@@ -36,17 +36,21 @@ export default function () {
         }
 
         let wrappedFunction = openmct.objects.get;
-        openmct.objects.get = function migrate(identifier) {
-            return wrappedFunction.apply(openmct.objects, [identifier])
+        openmct.objects.get = function migrate(identifier, fallback) {
+            return wrappedFunction.apply(openmct.objects, [identifier, fallback])
                 .then(function (object) {
-                    if (needsMigration(object)) {
-                        migrateObject(object)
-                            .then(newObject => {
-                                openmct.objects.mutate(newObject, 'persisted', Date.now());
-                                return newObject;
-                            });
+                    if (!object && fallback === undefined) {
```

Comment:
When creating a new domain object in our tree view, the couchDB object provider's get call returns `undefined` for a new object. This is when we need to get the cached new object and save it to CouchDB. That's where the fallbackProvider comes in.
I'm not sure how to fix this correctly... :/ Help!